android-components/components/feature/search/src/main/java/mozilla/components/feature/search/telemetry/SERPTelemetryJsonParser.kt

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.search.telemetry
+
+import android.content.Context
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import mozilla.components.support.base.log.logger.Logger
+import mozilla.components.support.ktx.android.org.json.toList
+import mozilla.components.support.remotesettings.RemoteSettingsClient
+import mozilla.components.support.remotesettings.RemoteSettingsResult
+import mozilla.components.support.remotesettings.readOrFetch
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.File
+
+/**
+ * Parse SERP Telemetry json from remote config.
+ */
+
+class SERPTelemetryJsonParser(context: Context) {
```

Comment:
nit: I try to use mixed case letters with acronyms based on their length as suggested in this [SO](https://stackoverflow.com/questions/2236807/java-naming-convention-with-acronyms)
```suggestion
class SerpTelemetryJsonParser(context: Context) {
```

Reply:
Just to note that Desktop does tend to use full caps for acronyms, which is where this has come from. I don't think it matters too much either way though.